modules/TestCasesRepository/TestCasesRouter.js

```diff
@@ -7,18 +7,35 @@ const testCasesRepository = new TestCasesRepository();
 
 // Define routes for the TestCaseRouter
 TestCaseRouter.get('/', (req, res) => {
-    res.send(testCasesRepository.getTestCases());
+    res.send(testCasesRepository.getTestCasesBySuite());
+});
+TestCaseRouter.get('/', (req, res) => {
+    res.send(testCasesRepository.getTestSuites());
 });
```

Comment:
W jednym routerze podpiąłeś tą samą metodę 'GET' na tej samej ścieżce '/' -> co spowoduje że tylko ten pierwszy będzie uruchamiany, ten drugi nigdy się nie wykona. Ponownie - zalecam stworzenie nowego routera pod to issue. Resztę sprawdzę po poprawkach.